bin/combine_grating_spectra

```diff
@@ -178,15 +178,22 @@ def split_II_to_I( instk, arfs, rmfs, outroot, verbose, clobber ):
         outs = [delme(x) for x in glob.glob( tmproot+"_?eg_??.pha")]
         bkgs = [delme(x) for x in glob.glob( tmproot+"_?eg_??_bkg.pha")]
 
+        outs.sort()
         retsrc.extend(outs)
 
         if ( 0 == len(bkgs) ):
             retbkg.extend([None]*len(outs))
-        else:
+        elif len(bkgs) == len(outs):
+            bkgs.sort()
             retbkg.extend(bkgs)
-            retbkg.sort()
-
-    retsrc.sort()
+        else:  # should never happen but just in case
+            raise IOError("Mismatch in number of source and background PHA files")
+
+    for sss,bbb in zip(retsrc,retbkg):
+        if bbb:
+            sroot = sss.rstrip(".pha")
+            broot = bbb.rstrip("_bkg.pha")
+            assert sroot == broot, "Something is messed up in sort/bkg matching"
```

Comment:
If this were to happen I assume there's nothing a user could do (e.g. can they explicitly list the files instead; it's too early for me to trace through to know exactly where this is happening to know if any of this is potentially user-controlled)?

Reply:
Correct.  Basically that would be a bug in `tgsplit`.  It should either produce no background or as many background as source files ; if there's a diff then we have to go back `tgsplit` and figure out what's wrong.

Reply:
So, in other words it's not worth cleaning up the error message.

Reply:
I don't this so.